src/subsystems/php/builders/simple/default.nix

```diff
@@ -76,6 +76,25 @@
         args.packages
       );
 
+    # Unpack dependency
+    cleanDependency = name: version:
```

Comment:
This adds an extra derivation for each dependency. Could this be implemented differently?

Reply:
Not sure. I need to create `composer.json` if it does not exist in the package sources.
Maybe adding like an `afterUnpack` hook to the `getSource`, or skipping `getSource` and using `getSourceSpec` to invoke the fetcher more directly and squeeze in the hook somewhere there.
What are your thoughts?

Reply:
Adding logic to the fetcher itself won't work, as it changes the output and therefore leads to a hash mismatch.

Yes, I think you could override the derivation returned by getSource.
Something like: `(getSource name version).overrideAttrs (old:{postUnpack = "";})`
But this will break for `path` sources which are not a derivation. So that case would have to be treated differently.

Still, it all sounds a bit hacky, so maybe it's better to keep the current implementation until we find a better solution.

Reply:
Sounds good

Reply:
`getSource` does not have `overrideAttrs` or `override`, seems like it returns the path directly and not a derivation, not sure.